reactioncommerce/reaction
# WIP #4516: Move formatPhoneNumber (and libphonenumber-js) server-side to reduce client bundle

Resolves #4516 
Impact: **minor**  
Type: **performance**

## Issue
libphonenumber-js was being included in the client bundle. It's only use is in `formatPhoneNumber`.

## Solution
Turn `formatPhoneNumber` into an async function that dynamically imports libphonenumber-js. In a separate PR, update the plugin that uses it (SMS in Reaction Contrib repo)

## Breaking changes
None


## Testing
1. Add the following code to /client/api/index.js:
import { formatPhoneNumber } from "/lib/api/helpers";
async function formatTest() {
  const num = await formatPhoneNumber("5555555555", "US");
  console.log("Formated phone number:", num);
}
formatTest();
2. Start the app and confirm you see in browser console: Formated phone number: 15555555555
3. Confirm bundle size is improved by running visualizer: `export TOOL_NODE_FLAGS='--max-old-space-size=4096' && METEOR_DISABLE_OPTIMISTIC_CACHING=1 meteor run --production --extra-packages bundle-visualizer`

## Before
![libphone-before](https://user-images.githubusercontent.com/849570/43793115-9f788766-9a48-11e8-8b51-825273c46c1f.jpg)

## After
![libphone-after](https://user-images.githubusercontent.com/849570/43793120-a3415d8c-9a48-11e8-884c-0bdef5c9d71b.jpg)
